test/gtest/ucs/test_async.cc

```diff
@@ -595,13 +596,11 @@ UCS_TEST_P(test_async, warn_block) {
         }
     }
 
-    int warn_count = m_warnings.size();
-    for (int i = 0; i < warn_count; ++i) {
-        UCS_TEST_MESSAGE << "< " << m_warnings[i] << " >";
-    }
-
     if (GetParam() != UCS_ASYNC_MODE_POLL) {
-        EXPECT_GE(warn_count, 1);
+        if (!m_warnings.empty()) {
+            UCS_TEST_MESSAGE << "< " << m_warnings.back() << " >";
```

Comment:
@brminich instead, need to clear warnings before each test in https://github.com/openucx/ucx/blob/master/test/gtest/common/test.cc#L222

Reply:
@yosefe, #4282